### PR TITLE
Replace custom AppInstance with Spring Cloud ServiceInstance

### PR DESCRIPTION
**Pull request by tgeens**
This PR is a first step towards using Spring Cloud Kubernetes Client Discovery instead of our own custom implementation.
It replaces our custom `AppInstance` by the Spring Cloud `ServiceInstance` - the abstraction from Spring Cloud Commons.

### CODE REVIEW

File: src/main/java/eu/xenit/alfred/content/gateway/GatewayApplication.java [hunk 1]
```diff
@@ -149,12 +152,23 @@ public ServiceTracker serviceTracker(ApplicationEventPublisher publisher, RouteL
 
         @Bean
         OpaQueryProvider opaQueryProvider(ServiceTracker serviceTracker) {
+            // TARGET ARCH: get application-id from request attributes, not from the service-tracker
             return request -> serviceTracker
-                    .findServices(s -> s.hostname().equals(request.getURI().getHost()))
+                    .findServices(s -> {
+                        var appId = s.getMetadata().get("app.contentgrid.com/application-id");
```
Comment (rschev) [inline src/main/java/eu/xenit/alfred/content/gateway/GatewayApplication.java:158]: Having to juggle these strings so deep in here looks less pleasant than how AppService used to expose it... Is it worth it having something else that extends ServiceInstance and gives better accessors, or have some static methods that do this (but all in one place so the label strings are only in one place)?

Comment (tgeens) [reply src/main/java/eu/xenit/alfred/content/gateway/GatewayApplication.java:158]: that's only temporary - the PR after the next cleans it up again ;-)

File: src/main/java/eu/xenit/alfred/content/gateway/servicediscovery/KubernetesServiceDiscovery.java [hunk 2]
```diff
@@ -3,40 +3,59 @@
...
             .build();
 
+    // TODO this should by a bean-init method
     @Override
     public void discoverApis() {
-        client.services().inNamespace(namespace).withLabelSelector(selector).watch(new Watcher<Service>() {
+        this.watch = client.services()
+                .inNamespace(namespace)
+                .withLabelSelector(selector)
+                .watch(new Watcher<Service>() {
```
[the remark this answers is not in the document]
Comment (tgeens) [reply src/main/java/eu/xenit/alfred/content/gateway/servicediscovery/KubernetesServiceDiscovery.java:42]: not yet no

File: src/main/java/eu/xenit/alfred/content/gateway/GatewayApplication.java (hunk 1 above)
Comment (tgeens) [reply src/main/java/eu/xenit/alfred/content/gateway/GatewayApplication.java:158]: maybe I should try and bring that forward into this PR

**Review by rschev — APPROVED** [review]

**Pull request merged**